packages/@okta/vuepress-site/code/vue/okta_vue_sign-in_widget/index.md

```diff
@@ -0,0 +1,358 @@
+  data () {
+    return {
+      claims: []
+    }
+  },
+  async created () {
+    this.claims = await Object.entries(await this.$auth.getUser()).map(entry => ({ claim: entry[0], value: entry[1] }))
+  }
+}
+</script>
+```
+
+### `/login`
+
+This route hosts the Sign-In Widget and redirects if the user is already logged in. If the user is coming from a protected page, they'll be redirected back to the page upon login.
+
+You should have already created `src/components/Login.vue` at the beginning of this guide.
+
+### `/login/callback`
+
+The component for this route (LoginCallback) comes with `@okta/okta-vue`. It handles token parsing, token storage, and redirecting to a protected page if one triggered the login.
+
+### Connect the Routes
+
+This example is using Vue Router. Replace the code in `src/router/index.js` with the following:
+
+```js
+import Vue from 'vue'
+import VueRouter from 'vue-router'
+import Auth from '@okta/okta-vue'
+import HomeComponent from '@/components/Home'
+import LoginComponent from '@/components/Login'
+import ProfileComponent from '@/components/Profile'
+
+Vue.use(VueRouter)
+Vue.use(Auth, {
+  issuer: 'https://{yourOktaDomain}/oauth2/default',
+  clientId: '{clientId}',
+  redirectUri: window.location.origin + '/login/callback',
+  scopes: ['openid', 'profile', 'email']
+})
+
+const router = new VueRouter({
+  mode: 'history',
+  routes: [
+    {
+      path: '/',
+      component: HomeComponent
+    },
+    {
+      path: '/login',
+      component: LoginComponent
+    },
+    {
+      path: '/login/callback',
+      component: Auth.handleCallback()
+    },
+    {
+      path: '/profile',
+      component: ProfileComponent,
+      meta: {
+        requiresAuth: true
+      }
+    }
+  ]
+})
+
+const onAuthRequired = async (from, to, next) => {
+  if (from.matched.some(record => record.meta.requiresAuth) && !(await Vue.prototype.$auth.isAuthenticated())) {
+    // Navigate to custom login page
+    next({ path: '/login' })
+  } else {
+    next()
+  }
+}
+
+router.beforeEach(onAuthRequired)
+
+export default router
+```
+
+Make sure to replace the `{...}` placeholders with values from your OIDC app on Okta.
+
+## Start your app
+
+Finally, start your app:
+
+```bash
+npm run serve
+```
+
+## Conclusion
+
+You have now successfully authenticated with Okta! Now what? With a user's `id_token`, you have basic claims for the user's identity. You can extend the set of claims by modifying the `scopes` to retrieve custom information about the user. This includes `locale`, `address`, `groups`, and [more](/docs/reference/api/oidc/).
+
+Want to learn how to use the user's `access_token`? Check out our <a href='/docs/guides/sign-into-spa/vue/before-you-begin/' data-proofer-ignore>Vue how to guide</a> to learn about protecting routes on your server, validating the `access_token`, and more!
+
+## Support
+
+Have a question or see a bug? Post your question on [Okta Developer Forums](https://devforum.okta.com/).
```

Comment:
```suggestion
Have a question or see a bug? Post your question on the [Okta Developer Forums](https://devforum.okta.com/).
```

Reply:
This matches what Angular and React have. Should I update those too?